keanu-project/src/main/java/io/improbable/keanu/model/linear/LinearRegression.java

```diff
@@ -0,0 +1,68 @@
+package io.improbable.keanu.model.linear;
+
+import java.util.Arrays;
+
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.ConstantVertex;
+import io.improbable.keanu.vertices.VertexLabel;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+
+public class LinearRegression implements LinearModel {
+
+    private static final double DEFAULT_MU = 0.0;
+    private static final double DEFAULT_SIGMA = 2.0;
+
+    private final BayesianNetwork net;
+    private final DoubleTensor y;
+
+    public LinearRegression(DoubleTensor x, DoubleTensor y) {
```

Comment:
A bit more of a design question here: Following the idea of 'pre-defined' models in Keanu would we perhaps want to decouple the model design from the data that goes in? This would be more similar to the way models are initialised in scikit-learn and would probably look something like this:
```
Model model = new LinearRegression()
DoubleTensor x = some data
DoubleTensor y = some labels
model.fit(x, y)
```
At the same time, the model structure (i.e. number of coefficients) is determined by the data that comes in. This would justify the currently implementation of the class. Or instead we'd have to only create the priors on the fly when the `fit` function is called.